docs/developers/build-starrocks/build_starrocks_on_ubuntu.md

```diff
@@ -0,0 +1,70 @@
+### Prerequisite
+```
+sudo apt-get install automake binutils-dev bison byacc ccache cmake flex libiberty-dev libtool maven zip -y
+```
+
+### Compiler
+Check the compilers(GCC, Python, JDK) are installed in the machine, if not install them
+```
+sudo apt-get install gcc g++ default-jdk python3 python-is-python3 -y
+```
+
+### Improve the compile speed
+The default compiling paralleilsim equals to the **CPU Cores / 4**. If you want to improve the compile speed. You can improve the paralleilsim. Suppose you have 32 Cpu cores, want to use 24 cores to compile.
+
+```
+./build.sh -j 24
+```
+
+### FAQ
+1. Failed to build `aws_cpp_sdk` in the Ubuntu 20.04.
+```
+Error: undefined reference to pthread_create
+```
+
+Apparently Threads::Threads is an interface library on some system. I encountered the same error with CMake 3.10.2, and applying the following patch fixed the issue.
+```
+diff --git a/cmake/dependencies.cmake b/cmake/dependencies.cmake
```

Comment:
Why not provide a build-ubuntu.sh script directly, but let the user make a patch?

Reply:
The problem comes from cmake's version. I think if I upgrade the cmake, the problem will dismiss.